jib-core/src/test/java/com/google/cloud/tools/jib/http/WithServerConnectionTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.http;
+
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link Connection} using an actual local server. */
+public class WithServerConnectionTest {
+
+  private static class WebServer implements Closeable {
+
+    private ServerSocket serverSocket;
+    private boolean stop;
+
+    private WebServer() throws IOException {
+      serverSocket = new ServerSocket(8087);
```

Comment:
Should we use a 5-digit port here for less chance of collisions?

Reply:
I'll just let it pick up any available port.